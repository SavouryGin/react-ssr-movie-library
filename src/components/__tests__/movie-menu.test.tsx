import MovieMenu from 'components/movie/menu';
import React from 'react';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';

const testProps = {
  className: 'test-class',
  onClose: jest.fn(),
  editMovie: jest.fn(),
  deleteMovie: jest.fn(),
};

describe('MovieMenu component:', () => {
  beforeEach(() => {
    render(<MovieMenu {...testProps} />);
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<MovieMenu {...testProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls the editMovie handler', () => {
    // arrange
    const editButton = screen.getByText('Edit');

    // act
    userEvent.click(editButton);

    // assert
    expect(testProps.editMovie).toHaveBeenCalledTimes(1);
  });

  it('calls the deleteMovie handler', () => {
    // arrange
    const deleteButton = screen.getByText('Delete');

    // act
    userEvent.click(deleteButton);

    // assert
    expect(testProps.deleteMovie).toHaveBeenCalledTimes(1);
  });

  it('calls the onClose handler', () => {
    // arrange
    const closeButton = screen.getByRole('button');

    // act
    userEvent.click(closeButton);

    // assert
    expect(testProps.onClose).toHaveBeenCalledTimes(1);
  });
});
