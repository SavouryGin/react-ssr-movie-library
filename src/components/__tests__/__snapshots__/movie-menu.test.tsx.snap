// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieMenu component: matches the snapshot 1`] = `
<DocumentFragment>
  <menu
    class="test-class"
    role="menu"
  >
    <button
      class="undefined"
      type="button"
    >
      <span
        class=""
      />
    </button>
    <li
      title="Edit this movie"
    >
      Edit
    </li>
    <li
      title="Delete this movie"
    >
      Delete
    </li>
  </menu>
</DocumentFragment>
`;
